### PR TITLE
Feat/outlier filter & flatter power forgetting curve & new pretrain

### PR DESCRIPTION
**Pull request by L-M-Sherlock**
- [x] Apply outlier filter to other models
- [x] Update FSRS-rs

### CODE REVIEW

**Comment by Expertium** [comment]
Just a reminder that this should be v4.5, a new entry. Don't just change v4.
To clarify, if only pretrain was changed, then I would agree that this should be called v4. But since one of the main formulas has been changed, this should be called v4.5.

**Comment by L-M-Sherlock** [comment]
I know about that. Due to a mistake, I have to re-run all the models😭. And I need to compare v4 with v4.5 by using the same outlier filter, so I need to run the script twice.

**Comment by Expertium** [comment]
Minor nitpick: it should be called "FSRS v4.5", not "FSRS-4.5". Since previous versions have the letter v, let's keep it consistent.

**Comment by L-M-Sherlock** [comment]
Due to the semantic version rule, I have used v4.5 in fsrs-optimizer: https://github.com/open-spaced-repetition/fsrs-optimizer/releases/tag/v4.5.0. So it's ambiguous.

File: analysis.py
Comment (Expertium) [inline analysis.py:1]: I don't think that's a good idea. I've mentioned before that mode is sensitive to clustering. If we don't remove users who have parameters that are very close to their defaults, there will be clusters and the mode will just return the defaults.

Comment (L-M-Sherlock) [reply analysis.py:1]: If we check all parameters, the user will be removed even if only one parameter is equal to the default parameter.

Comment (Expertium) [reply analysis.py:1]: Yes, but it's better than using a mode which just returns the same default parameters.

Comment (L-M-Sherlock) [reply analysis.py:1]: Done.

**Pull request merged**